streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -713,6 +727,48 @@ boolean runLoop() {
         return true;
     }
 
+    // visible for testing
+    void maybeGetClientInstanceIds() {
```

Comment:
I think there is a risk here that if the client calls `clientInstanceIds` faster than we are going around the poll loop, we will have an every-growing number of futures to complete, and in each iteration we can only complete one. We may also run out of time this way.